plugins/flatpak/gs-plugin-flatpak.c

```diff
@@ -728,6 +728,14 @@ gs_plugin_app_remove (GsPlugin *plugin,
 		gs_flatpak_error_convert (error);
 		return FALSE;
 	}
+
+	/* If the app is still in REMOVING it's probably because it was installed
+	 * from a .flatpak bundle and is therefore no longer available (unless the
+	 * bundle is still around).
+	 */
+	if (gs_app_get_state (app) == AS_APP_STATE_REMOVING)
```

Comment:
Maybe also check whether `app` comes from a bundle here?

Reply:
Agree that's a good idea

Reply:
But sadly we can't check if `gs_flatpak_app_get_file_kind (app) == GS_FLATPAK_APP_FILE_KIND_BUNDLE` because that's only true for the `GsApp` created when the bundle is opened, not the `GsApp` created for the installed app.

One thing we could check is if the associated remote has an empty `url`. But then there's the question of what would happen for bundles which include a URL. I think those should have the app picked up by `gs_flatpak_add_apps_from_xremote()` but it depends on if they serve appstream data so I'm not sure.

Reply:
I see - just wondering if this could break uninstallation for other cases, have you tested it? Does the Install button shows up right after an app (not bundle) gets uninstalled? Can you reinstall an app right after uninstalling it?

What happens in the UI when uninstalling a bundle?

Reply:
This code is after the uninstall transaction completes and the remotes have been scanned again for apps (`gs_flatpak_refresh()`) so it shouldn't apply to apps that were not from bundles. Even so, I'll do some more testing and get back to you.

Reply:
For an app installed from a bundle, there's a spinner while it's being removed, then that goes away and after a few seconds the details page is replaced by the message "Unable to find org.gnome.Fractal" which seems like reasonable behavior.

For an app installed from the Internet, the "Removing" spinner goes away and after a few seconds a "Download" button appears as you'd expect. (There is an upstream MR to make the spinner spin longer: https://gitlab.gnome.org/GNOME/gnome-software/merge_requests/367)

Reply:
> Can you reinstall an app right after uninstalling it?

Yes, for an app installed from the Internet, I was able to use the Download button successfully immediately after uninstalling.

Reply:
> For an app installed from a bundle, there's a spinner while it's being removed, then that goes away and after a few seconds the details page is replaced by the message "Unable to find org.gnome.Fractal" which seems like reasonable behavior.
> 
I guess that is reasonable, the other option would be to go back to the main page I guess, but that would probably make the user even more confused. Lets go with this as is and we can iterate later if needed.